packages/build-tools/src/steps/functions/download_build.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  BuildFunction,
+  BuildStepInput,
+  BuildStepInputValueTypeName,
+  BuildStepOutput,
+} from '@expo/steps';
+
+export function createDownloadBuildFunction(): BuildFunction {
+  return new BuildFunction({
+    namespace: 'eas',
+    id: 'download_build',
+    name: 'Download build',
+    inputProviders: [
+      BuildStepInput.createProvider({
+        id: 'build_id',
+        required: true,
+        allowedValueTypeName: BuildStepInputValueTypeName.STRING,
+      }),
+    ],
+    outputProviders: [
+      BuildStepOutput.createProvider({
+        id: 'artifact_path',
+        required: false,
+      }),
+    ],
+    command: `
+    BUILD_ID='\${ inputs.build_id }'
+    APPLICATION_ARCHIVE_URL="\${__API_SERVER_URL}v2/artifacts/eas/$BUILD_ID"
+
+    echo "Downloading build from $APPLICATION_ARCHIVE_URL"
+    if ! STORED_FILE_URL=$(curl --remote-name --location --fail -w %{url_effective} --silent --show-error "$APPLICATION_ARCHIVE_URL"); then
+      echo "File not available, skipping..."
+      exit 0
+    fi
```

Comment:
if we fail to download the file inside of the `eas/download_file` function shouldn't we fail? it seems to be the core thing this step does

proceeding with successful `eas/download_build` and having some workflows fail in later steps due to missing build afterward seems like a hard thing to debug

as a user, I would prefer it to fail straightaway